content/blogs/GSoC-2020-rapps-final-report.md

```diff
@@ -0,0 +1,88 @@
+[Pull Request 3108](https://github.com/reactos/reactos/pull/3108)
+
+### Stage 1
+
+During stage 1, I mainly focused on screenshot and icon support as well as various small improvements.
+
+#### work done:
+- Add support for screenshots. 
+
+- Improve icon support. 
+
+- Some other invisible improvements.
+
+### Stage 2
+
+During stage 2, I mainly focused on improving listview and related code. This turned out to be a pretty big refactor, and it improved the UI a bit.
+
+#### work done:
+- Move ToolBar and SearchBar to AppView, and remove "settings" and "exit" button.
+
+- Enable users to choose how to view apps using a ComboBox.
+
+- Some small bugfix and improvements. 
+
+- Some other invisible improvements and some refactoring.
+
+##### More detailed information about work I've done during stage 1 & 2 can be found here: [GSoC 202 RAPPS Stage 1 & 2](/blogs/gsoc-2020-rapps-stage12/)
+
+### Stage 3
+
+During stage 3, I mainly focused on improving command-line option for rapps, turning rapps into a command line app in order to output to console. Some work improving settings storage are also done.
+
+#### work done:
+
+- Using package name in command line instead of using software's name.
+
+- Turning rapps a command-line program. It will detach the console when it's asked to run in GUI mode （e.g. when no parameters being passed to it).
+
+- Add ```/?``` command to show basic usage of rapps commandline.
+
+- Add /find and /info command (for search and displaying software information)
+
+- some minor improvements. (Allow using ```-``` as option prefix, and so on)
+
+- Store settings in registery in a more human-friendly form instead of binary form.
+
+**screenshots**
+![RAPPS screenshot](/img/blogs/gsoc2020-rapps-final-report.png)
+
+## work left
+because of my limited time and ability, there's still some work left undone. Some are left by me, some are bugs/imperfection I discovered and don't have time to fix. I will list all of them here.
+
+
+- Currently, only first screenshot will be displayed. It should be displayed somewhere.
+
+- Clicking links in richedit using left button with ```Ctrl``` pressed, should just open the browser instead of popup a menu.
+
+- Code about downloading apps need a refactor. It should reuse code in ```AsyncInet.cpp```.
+
+- Currently, downloaded apps will be stored all in a same folder using filename extracted from URL. (code can be found in ```loaddlg.cpp```) there are some problems with this design. 
+
+    - URL may not have a filename. for example, see twsynth.txt （probably in this case we should retrieve filename form HTTP header?)
+    - Different apps should be stored in their respective folders, otherwise if filename in URL is the same, it could lead to a disaster.
+    - URL can be quite long, and it may contain illegal character in it. we should make sure it's shorter than ```MAX_PATH``` and no illegal character in it.
```

Comment:
```suggestion
    - The URL can be quite long, and it may contain illegal characters. We should make sure it's shorter than ```MAX_PATH``` and has no illegal characters in it.
```